### PR TITLE
Add Cache-Control header

### PR DESCRIPTION
**Pull request by seanwiseman**
Addresses [#4382](https://github.com/elifesciences/issues/issues/4382).

### CODE REVIEW

File: app/core/middleware.py [hunk 1]
```diff
@@ -46,3 +47,33 @@ def middleware(request: Request):
...
+        private_headers = {
+            'private': True,
+            'max-age': 0,
+            'must-revalidate': True,
+        }
+
+        response = get_response(request)
+
+        if request.META.get(settings.AUTHORIZATION_PREVIEW_HEADER, False):
+            cache_headers = private_headers
```
Comment (giorgiosironi) [inline app/core/middleware.py]: this is a bit magic, if it's someone who can preview then don't cache. Works for `journal--preview`, may be good enough but it's a strange decision

File: app/digests/tests/test_cache_control.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+@pytest.mark.django_db
+def test_has_public_cache_control_when_unauthenticated(client: Client):
+    response = client.get(DIGESTS_URL,
+                          **{'ACCEPT': settings.DIGESTS_CONTENT_TYPE})
+    assert response.status_code == 200
+    assert response['cache-control'] == 'public, max-age=300, ' \
+                                        'stale-while-revalidate=300, ' \
+                                        'stale-if-error=86400'
```
[the remark this answers is not in the document]
Comment (thewilkybarkid) [reply app/digests/tests/test_cache_control.py]: Should be `must-revalidate, no-cache, no-store, private`.

File: app/core/middleware.py (hunk 1 above)
Comment (giorgiosironi) [reply app/core/middleware.py]: For now this part should be fine as its underlying intention is: if you show private content, do not cache it anywhere (especially in public caches)

**Comment by giorgiosironi** [comment]
Related: are all the `Vary` headers as we expect? Kong will add `Vary: Authorization`

**Comment by seanwiseman** [comment]
> Related: are all the Vary headers as we expect? Kong will add Vary: Authorization

The responses currently contain `Vary Accept, Cookie` (which are just defaults) it does not contain `Accept-Encoding`.

Presumably need to add `Accept-Encoding` ?

**Comment by giorgiosironi** [comment]
Yes, I can confirm `Vary: Accept-Encoding` is there on other listings like `/articles` so it should probably be added here too.

File: app/core/middleware.py [hunk 3]
```diff
@@ -75,6 +76,8 @@ def middleware(request: Request):
 
             patch_cache_control(response, **cache_headers)
 
+        patch_vary_headers(response, ['Accept', 'Accept-Encoding'])
```
Comment (thewilkybarkid) [inline app/core/middleware.py]: Is Django handling compression? (Normally done at Nginx etc?)

Comment (seanwiseman) [reply app/core/middleware.py]: That would explain it then, seemed strange. Will remove.

**Pull request merged**